packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -237,19 +237,34 @@ async function getOutlookVersion(): Promise<string | undefined> {
   }
 }
 
-async function getOutlookExePath(): Promise<string> {
+async function getWXPOExePath(app: OfficeApp): Promise<string> {
+  let HostApp: string = "";
   try {
-    const OutlookInstallPathRegistryKey: string = `HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\OUTLOOK.EXE`;
-    const key = new registry.RegistryKey(`${OutlookInstallPathRegistryKey}`);
-    const outlookExePath: string | undefined = await registry.getStringValue(key, "");
-
-    if (!outlookExePath) {
-      throw new Error("Outlook.exe registry empty");
+    switch (app) {
+      case OfficeApp.Excel:
+        HostApp = "EXCEL";
+        break;
+      case OfficeApp.Outlook:
+        HostApp = "OUTLOOK";
+        break;
+      case OfficeApp.Word:
+        HostApp = "WINWORD";
+        break;
+      case OfficeApp.PowerPoint:
+        HostApp = "POWERPNT";
+        break;
```

Comment:
Should always have a default case in the switch.

Reply:
Thanks for Darren's heads up!

Reply:
I have tested the registryKey, it can lunch WXPO hosts correctly.

Reply:
Got it. Will change.